eva_cttv_pipeline/clinvar_to_evidence_strings.py

```diff
@@ -1,280 +1,314 @@
+                                                                                  record.measure_set_refs_list,
+                                                                                  record.observed_regs_list,
+                                                                                  rcv_to_gene_evidence_codes,
+                                                                                  cellbase_record,
+                                                                                  record.rs,
+                                                                                  trait.trait_counter,
+                                                                                  record.trait_refs_list,
+                                                                                  report)
+                elif allele_origin == 'somatic':
+                    evidence_string = evidence_strings.CTTVSomaticEvidenceString(trait.efo_list,
+                                                                                 record.clin_sig,
+                                                                                 record.clinvarRecord,
+                                                                                 ensembl_gene_id,
+                                                                                 ensembl_gene_id_uri,
+                                                                                 record.measure_set_refs_list,
+                                                                                 record.observed_regs_list,
+                                                                                 trait.trait_counter,
+                                                                                 record.trait_refs_list,
+                                                                                 record.con_type,
+                                                                                 report)
+                add_evidence_string(record, evidence_string, report)
+                report.evidence_list.append(
+                    [record.clinvarRecord.accession, record.rs, ','.join(trait.clinvar_trait_list),
+                     ','.join(trait.efo_list)])
+                report.counters["n_valid_rs_and_nsv"] += (record.clinvarRecord.get_nsv(mappings.rcv_to_nsv) is not None)
+                report.counters["n_more_than_one_efo_term"] += (len(trait.efo_list) > 1)
+                report.traits.update(set(trait.efo_list))
+                report.ensembl_gene_id_uris.add(ensembl_gene_id_uri)
+
+        if record.n_ev_strings_per_record > 0:
+            report.counters["n_processed_clinvar_records"] += 1
+            if record.n_ev_strings_per_record > 1:
+                report.counters["n_multiple_evidence_strings"] += 1
+
+    report.write_output(dir_out)
+
+    print(report)
```

Comment:
I feel keen to extract the `write_output` and `print` calls out from this function (probably into the `main` one) and make `clinvar_to_evidence_strings` return the Report object. This would split the actual generation of the report from any IO, and remove the `dir_out` argument from the list of arguments.